### ISSUE TITLE
Starting multiple separate i3 instances doesn't work

### ISSUE THREAD

**Issue by SergeyKasmy** [i3/i3#4108]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
Starting a second instance of i3, e.g. in VNC, Xephyr, or just a separate X display, doesn't create a separate ipc socket. Instead it "steals" the socket from the first i3 instance, breaking all programs that rely on it, e.g. polybar. After it's stopped, it deletes the socket with it, leaving the first i3 instance unreachable, without even being able to exit it gracefully. It does create a separate log file but not a socket. 

Before starting a separate i3 instance:
```
$ ls -l /run/user/1000/i3/
.rw-r--r-- user user 0 B Mon May 25 04:21:47 2020  errorlog.5807  
srwxr-xr-x user user 0 B Mon May 25 04:21:47 2020  ipc-socket.5807
```

After starting it:
```
$ ls -l /run/user/1000/i3/
.rw-r--r-- user user 0 B Mon May 25 04:21:47 2020  errorlog.5807  
.rw-r--r-- user user 0 B Mon May 25 04:21:47 2020  errorlog.13051  
srwxr-xr-x user user 0 B Mon May 25 04:21:47 2020  ipc-socket.5807
```

After stopping it:
```
$ ls -l /run/user/1000/i3/
.rw-r--r-- user user 0 B Mon May 25 04:21:47 2020  errorlog.5807  
```

## Expected Behavior
Separate i3 instances should not interfere with each other. They should create their own logs and ipc sockets and remove them by themselves.

## Reproduction Instructions
The easiest way to experience the bug is to use Xephyr:
1. Start your PC, login & start i3
1. Open a terminal
2. Run Xephyr, e.g. `Xephyr -ac -screen 1920x1080 :1`
3. Open a different terminal
4. Run i3, e.g. `env DISPLAY=:1 i3 &`
5. Notice that it "stole" the socket from the first i3 instance
6. Stop the i3 Xephyr instance, e.g. `kill %1`
7. Notice that it deleted the ipc socket and there isn't a way to reach the first i3 instance anymore 

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: Binary i3 version:  4.18.1 (2020-04-22)
</pre>

<pre>
- Linux Distribution & Version: Arch Linux
- Are you using a compositor (e.g., xcompmgr or compton): compton-tryone
</pre>

**Comment by i3bot** [comment]
I don’t see a link to logs.i3wm.org. Did you follow https://i3wm.org/docs/debugging.html? (In case you actually provided a link to a logfile, please ignore me.)

**Comment by SergeyKasmy** [comment]
@i3bot I'm not able to provide a log right now but I'll provide it tomorrow when I'll be using my PC once more

**Comment by orestisfl** [comment]
You have to unset `I3SOCK`.

This happens after https://github.com/i3/i3/pull/3484.

**Comment by SergeyKasmy** [comment]
Seems kinda unintuitive to me. I get why somebody would want to have `$I3SOCK` but, when started, shouldn't i3 overwrite it if it already exists? Having to unset the variable every time I want to start vncserver or Xephyr seems really unnecessary and cumbersome

**Comment by orestisfl** [comment]
Yes, this is something I don't like as well. I have a patch that doesn't
use I3SOCK if it exists but I didn't have the time to test it yet.

**Comment by MatteCarra** [comment]
I still have this issue on i3 4.20.
Any update? Thanks

**Comment by orestisfl** [comment]
Superseded by https://github.com/i3/i3/issues/4381
